MonoGame.Framework.Content.Pipeline/Graphics/Font/SharpFontImporter.cs

```diff
@@ -9,117 +9,157 @@
+                    }
+                }
+                glyphIndexes.RemoveAll(p => p.Value != 0);
+                //For each character not included in the font load the glyph with index 0
+                face.LoadGlyph(0, LoadFlags.Default, LoadTarget.Normal);
+                face.Glyph.RenderGlyph(RenderMode.Normal);
+                foreach (var character in glyphIndexes)
+                {
+                    //var glyph = ImportGlyph(character.Key, face);
+                    var glyph = RetrieveGlyphData(character.Key, face);
+                    glyphList.Add(glyph);
+                }
+                Glyphs = glyphList;
+
+                // Store the font height.
+                LineSpacing = (maxAscender + maxDescender) >> 6;
+
+                // The height used to calculate the Y offset for each character.
+                YOffsetMin = -maxAscender >> 6;
+            }
             finally
             {
-				if (face != null)
-					face.Dispose();
-				if (lib != null)
+                if (face != null)
+                    face.Dispose();
+                if (lib != null)
                 {
-					lib.Dispose();
-					lib = null;
-				}
-			}
-		}
+                    lib.Dispose();
+                    lib = null;
+                }
+            }
+        }
 
 
-		// Attempts to instantiate the requested GDI+ font object.
-		private Face CreateFontFace(FontDescription options, string fontName)
-		{
-			try
+        // Attempts to instantiate the requested GDI+ font object.
+        private Face CreateFontFace(FontDescription options, string fontName)
+        {
+            try
             {
-				const uint dpi = 96;
-				var face = lib.NewFace(fontName, 0);
-				var fixedSize = ((int)options.Size) << 6;
-				face.SetCharSize(0, fixedSize, dpi, dpi);
-
-				if (face.FamilyName == "Microsoft Sans Serif" && options.FontName != "Microsoft Sans Serif")
-					throw new PipelineException(string.Format("Font {0} is not installed on this computer.", options.FontName));
-
-				return face;
-
-				// A font substitution must have occurred.
-				//throw new Exception(string.Format("Can't find font '{0}'.", options.FontName));
-			}
-			catch
-			{
-				throw;
-			}
-		}
-
-		// Rasterizes a single character glyph.
-		private Glyph ImportGlyph(char character, Face face)
-		{
-			uint glyphIndex = face.GetCharIndex(character);
-			face.LoadGlyph(glyphIndex, LoadFlags.Default, LoadTarget.Normal);
-			face.Glyph.RenderGlyph(RenderMode.Normal);
-
-			// Render the character.
+                const uint dpi = 96;
+                var face = lib.NewFace(fontName, 0);
+                var fixedSize = ((int) options.Size) << 6;
+                face.SetCharSize(0, fixedSize, dpi, dpi);
+
+                if (face.FamilyName == "Microsoft Sans Serif" && options.FontName != "Microsoft Sans Serif")
+                    throw new PipelineException(string.Format("Font {0} is not installed on this computer.", options.FontName));
```

Comment:
I'm not completely sure, but I think this is done because in windows that is the default font when a font is not found or a character is not included in the requested font.
For example, in XNA, if you use any of the fonts from the redistributable font pack and include characters beyond the 128, instead of blank characters you get the ones from "Microsoft Sans Serif".